### PR TITLE
update figure exporters

### LINKED ISSUE #36: [add_feature] export figures
Related code:
```
export_figures = False
```
and
```
if export_figures == True:
	plt.savefig("figures/xx.pdf")
```
where ``xx`` is the figure’s name.

export_figures name lists for each chapter:

ch_intro
- [x] crude_oil_2019.pdf
- [x] commercial_aircraft_2019_1.pdf
- [x] simplex_1.pdf
- [x] heavy_tailed_draws.pdf
- [x] ccdf_comparison_1.pdf
- [x] empirical_powerlaw_plots_firms_forbes.pdf
- [x] zeta_1.pdf
- [x] networkx_basics_1.pdf
- [ ] financial_network_analysis_visualization.pdf
- [x] financial_network_analysis_centrality.pdf
- [x] commercial_aircraft_2019_2.pdf
- [x] rand_graph_experiments_1.pdf
- [x] rand_graph_experiments_2.pdf

ch_production
- [ ] input_output_analysis_15.pdf
- [x] input_output_analysis_71.pdf
- [x] input_output_analysis_15_leo.pdf
- [x] input_output_analysis_15_shocks.pdf
- [x] input_output_analysis_15_ec.pdf
- [x] input_output_analysis_aus_114.pdf
- [x] input_output_analysis_15_omult.pdf
- [ ] input_output_analysis_15_fwd.pdf
- [x] input_output_analysis_15_up.pdf
- [ ] gdp_growth.pdf
- [x] input_output_analysis_15_katz.pdf

ch_opt
- [x] shortest_path_iter_1.pdf
- [x] betweenness_centrality_1.pdf
- [x] linear_programming_1.pdf
- [x] ot_figs_1.pdf
- [x] ot_large_scale_1.pdf

ch_mcs
- [x] markov_matrix_visualization.pdf
- [x] benhabib_mobility_mixing.pdf
- [x] quah_gdppc_prediction.pdf
- [x] simplex_2.pdf
- [x] simplex_3.pdf
- [x] benhabib_mobility_dists.pdf
- [x] benhabib_ergodicity_1.pdf

ch_fpms
- [x] fin_network_sims_1.pdf

appendix
- [x] func_types_1.pdf
- [x] func_types_2.pdf
- [x] three_fixed_points.pdf
- [x] complex_number.pdf
- [x] euclidean_convergence_1.pdf
- [x] span1.pdf
- [x] func_types_3.pdf
- [x] polyhedron1.pdf
- [x] saddle_1.pdf

### PR DESCRIPTION
**Pull request by shlff**
Hi  @jstac this PR fixes #36 .

### CODE REVIEW

**Comment by jstac** [comment]
@shlff , many thanks.

Is this ready for review?  Can you please confirm that you have clicked on the link https://6267d4684574fb1971819090--keen-wozniak-c08666.netlify.app/ above and looked through all the figures.

**Comment by shlff** [comment]
Thank you @jstac . This PR is not ready for review yet.

I have been verifying the filename-figure correspondences with those used in the textbook. I'll keep you posted for reviewing later.

File: code_book/appendix.md [hunk 1]
```diff
@@ -51,6 +52,8 @@ for f, ax, lb, ti in zip(funcs, axes, labels, titles):
     ax.set_xticks((0, 1))
     ax.set_yticks((0, 1))
 
+if export_figures == True:
```
Comment (mmcky) [inline code_book/appendix.md:55]: @shlff if `export_figures` is boolean I would suggest just using:

```python
if export_figures:
   <work>
```

Comment (jstac) [reply code_book/appendix.md:55]: Good idea, thanks @mmcky

**Comment by shlff** [comment]
Hi @jstac to ensure that figures generated from this repo are consistent with those used in the textbook there are some extra tasks to be completed before reviewing this PR:
- [ ] update the corresponding ``matplotlib`` configuration to the repo ``quantecon-book-networks`` (see [PR](https://github.com/QuantEcon/quantecon-book-networks/pull/1)) and update the corresponding imports in this repo
- [ ] fix the code generating the figure ``financial_network_analysis_visualization.pdf``
- [ ] fix the code generating figures ``input_output_analysis_15.pdf`` and ``input_output_analysis_15_fwd.pdf``

**Comment by shlff** [comment]
Thanks @jstac . As we discussed this PR is ready for review.

Now the only issue comes from generating the following 3 figures, which I would investigate and solve in another PR:
- [ ] financial_network_analysis_visualization.pdf
- [ ] input_output_analysis_15.pdf
- [ ] input_output_analysis_15_fwd.pdf

File: code_book/appendix.md [hunk 2]
```diff
@@ -413,16 +430,19 @@ ax.plot((4.8,), (-1.2,), 'ko')
...
-### A saddle point $(x^*, \theta^*)$ of the function $L$ (top left)
-
 ```{code-cell}
-fig = plt.figure(figsize=(12, 7))
-ax = plt.axes(projection ='3d')
+fig = plt.figure(figsize=(8.5, 6))
+
+## Top left Plot
+
+ax = fig.add_subplot(221, projection='3d')
```
Comment (shlff) [inline code_book/appendix.md:445]: With these two changes, we make the figure size consistent with the one used in the textbook and get rid of the sleep loops.

**Comment by jstac** [comment]
Great work @shlff , excellent job.

One other thing I noticed: in production, the time series for Canada in figures/gdp_growth.pdf is different to the textbook version (fig 2.11).  Could you please add that to the list above?  

Once you confirm I will merge this PR.

**Comment by shlff** [comment]
> Great work @shlff , excellent job.
> 
> One other thing I noticed: in production, the time series for Canada in figures/gdp_growth.pdf is different to the textbook version (fig 2.11). Could you please add that to the list above?
> 
> Once you confirm I will merge this PR.

Thanks for your careful check @shlff . I can confirm that, and I've added the figure ``gdp_growth.pdf`` to the list above, as well as #36 .

I will investigate the issues in these four figures next.

**Comment by shlff** [comment]
> Great work @shlff , excellent job.
> 
> One other thing I noticed: in production, the time series for Canada in figures/gdp_growth.pdf is different to the textbook version (fig 2.11). Could you please add that to the list above?
> 
> Once you confirm I will merge this PR.

Thanks for your good comments and careful check @jstac . I can confirm that, and I've added the figure ``gdp_growth.pdf`` to the list above, as well as #36 .

I will investigate the issues in these four figures next.

**Pull request merged**